### PR TITLE
Documentation gardening

### PR DESCRIPTION
**Pull request by Hirrolot**

### CODE REVIEW

File: src/requests/payload.rs [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 ///
-/// Simply speaking structs implementing this trait represent arguments of
-/// a telegram bot API method.
+/// Simply speaking, structures implementing this trait represent arguments of
+/// a Telegram bot API method.
 ///
-/// This trait provides some additional information needed for sending request
-/// to the telegram.
+/// Also, this trait provides some additional information needed for sending a
+/// request to Telegram.
```
Comment (WaffleLapkin) [inline src/requests/payload.rs:7]: maybe?
```suggestion
/// Also, this trait provides some additional information needed to send a
/// request to Telegram.
```

File: src/requests/payload.rs [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
 #[cfg_attr(all(docsrs, feature = "nightly"), doc(spotlight))]
 pub trait Payload {
-    /// Return type of the telegram method.
+    /// A return type of a Telegram method.
```
Comment (WaffleLapkin) [inline src/requests/payload.rs:10]:
```suggestion
    /// The return type of a Telegram method.
```

File: src/requests/request.rs [hunk 3]
```diff
@@ -23,19 +23,20 @@ pub trait Request: HasPayload {
      * use it before it's integrated in async/await
      */
 
-    /// Type of error that may happen during sending the request to telegram.
+    /// A type of an error that may happen while sending a request to Telegram.
```
Comment (WaffleLapkin) [inline src/requests/request.rs:26]:
```suggestion
    /// The type of an error that may happen while sending a request to Telegram.
```

File: src/requests/request.rs [hunk 4]
```diff
@@ -23,19 +23,20 @@ pub trait Request: HasPayload {
...
     type Err: std::error::Error + Send;
 
-    /// Type of future returned by [`send`](Request::send) method.
+    /// A type of the future returned by the [`send`](Request::send) method.
```
Comment (WaffleLapkin) [inline src/requests/request.rs:29]:
```suggestion
    /// The type of the future returned by the [`send`](Request::send) method.
```

File: src/requests/request.rs [hunk 5]
```diff
@@ -23,19 +23,20 @@ pub trait Request: HasPayload {
...
     type Send: Future<Output = Result<Output<Self>, Self::Err>> + Send;
 
-    /// Type of future returned by [`send_ref`](Request::send_ref) method.
+    /// A type of the future returned by the [`send_ref`](Request::send_ref)
+    /// method.
     ///
-    /// NOTE: it intentionally forbids borrowing from self
-    // though anyway we couldn't allow borrowing without GATs :sob:
+    /// Note: it intentionally forbids borrowing from `self` though anyway we
+    /// couldn't allow borrowing without GATs.
```
Comment (WaffleLapkin) [inline src/requests/request.rs:36]: I don't think we need to include this minor note in docs. But if we do, it's better to place it in braces or in a different sentence. IMHO.

File: src/requests/request.rs [hunk 6]
```diff
@@ -51,15 +52,15 @@ pub trait Request: HasPayload {
...
     /// This method is analogous to [`send`](Request::send), but it doesn't take
     /// the ownership of `self`. This allows to send the same (or slightly
     /// different) requests over and over.
     ///
-    /// _Also_ it is expected that calling this method is better than just
-    /// `clone`ing the requests. (because instead of copying all the data
-    /// and then serializing it, this method should just serialize the data)
+    /// Also, it is expected that calling this method is better than just
+    /// cloning requests. (Because instead of copying all the data
+    /// and then serializing it, this method should just serialize the data.)
```
Comment (WaffleLapkin) [inline src/requests/request.rs:63]: I haven't seen dots at the end of a sentence in braces before, are you sure it's mandatory?

File: src/requests/requester.rs [hunk 7]
```diff
@@ -22,43 +23,43 @@ pub trait Requester {
 
     type GetUpdates: Request<Payload = GetUpdates, Err = Self::Err>;
 
-    /// For telegram documentation see [`GetUpdates`]
+    /// For Telegram documentation see [`GetUpdates`].
```
Comment (WaffleLapkin) [inline src/requests/requester.rs:26]: Open an issue, I'll fix it later.

https://github.com/teloxide/teloxide-core/blob/e260f11558e5bfae89b4bfa58cd3d99a487da1ab/src/requests/requester.rs#L19-L21

**Review by WaffleLapkin — CHANGES_REQUESTED** [review]

File: src/adaptors.rs [hunk 8]
```diff
@@ -1,11 +1,10 @@
...
-//! Bot adaptors are very similar to [`Iterator`] adaptors —
-//! they are bots (implement [`Requester`]) which wrap other bots
-//! adding new functionality.
+//! Bot adaptors are very similar to the [`Iterator`] adaptors: they are bots
+//! wrapping other bots to alter existing or add new functionality.
 //!
-//! E.g. [`AutoSend`] allows `await`ing requests directly,
-//! without need to use `.send()`.
+//! E.g. [`AutoSend`] allows `await`ing requests directly, no need to to use
+//! `.send()`.
```
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/adaptors.rs:7]: By the way, have I put two `to`?

File: src/errors.rs [hunk 9]
```diff
@@ -8,24 +8,25 @@ use thiserror::Error;
 /// An error caused by downloading a file.
 #[derive(Debug, Error, From)]
 pub enum DownloadError {
-    /// Network error while downloading file from telegram.
+    /// Network error while downloading a file from Telegram.
     #[error("A network error: {0}")]
     NetworkError(#[source] reqwest::Error),
 
-    /// I/O error while writing file to destination.
+    /// An I/O error while writing a file to destination.
```
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/errors.rs:15]: It's my mistake, I'll fix it.

File: src/net/download.rs [hunk 10]
```diff
@@ -11,17 +11,15 @@ use tokio::io::{AsyncWrite, AsyncWriteExt};
...
-/// Download client.
-///
-/// This trait allows you to download files from telegram.
+/// A trait for downloading files from Telegram.
 pub trait Download<'w>
 /* FIXME(waffle): ideally, this lifetime ('w) shouldn't be here, but we can't help it without
  * GATs */
 {
-    /// Error returned by [`download_file`](Self::download_file)
+    /// An error returned from [`download_file`](Self::download_file).
```
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/net/download.rs:19]: Not sure, but this is consistent with other documentation strings (they use `from`).

File: src/payloads/mod.rs [hunk 11]
```diff
@@ -1,11 +1,14 @@
-//! Payloads - data types sended to relegram
+//! Request data sent to Telegram.
```
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/payloads/mod.rs:1]: It is already in the module's name.

File: src/requests/json.rs [hunk 12]
```diff
@@ -6,11 +6,9 @@ use crate::{
     RequestError,
 };
 
-/// Ready-to-send telegram request.
+/// A ready-to-send Telegram request whose payload is sent using [JSON].
```
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/requests/json.rs:9]: `JSON` is capitalised as Wikipedia tells: https://en.wikipedia.org/wiki/JSON.

File: src/requests/request.rs (hunk 6 above)
Comment (Hirrolot) [reply src/requests/request.rs:63]: Yes, see https://www.gutenberg.org/files/37134/37134-h/37134-h.htm#Page_33.

File: src/requests/json.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (WaffleLapkin) [reply src/requests/json.rs:9]: Well, yaml and xml are capitalized too by Wikipedia, but I think it's just annoying.

File: src/requests/request.rs (hunk 6 above)
Comment (WaffleLapkin) [reply src/requests/request.rs:63]: Feels strange to me, but ok, thanks for the link.

File: src/requests/json.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (Hirrolot) [reply src/requests/json.rs:9]: reqwest writes the capitalised version too.

File: src/requests/request.rs (hunk 5 above)
Comment (Hirrolot) [reply src/requests/request.rs:36]: Yes, I'll just make this comment private.

**Review by WaffleLapkin — APPROVED** [review]
I'm still thinking that it's easier to read `json` (as opposed to `JSON`). But it's a minor thing so ¯\\\_(ツ)\_/¯

**Pull request merged**